From 28fe60f42007b3d1620cf347e4262bdaf81b64d8 Mon Sep 17 00:00:00 2001
From: kboeluek <kboeluek@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 22 Jan 2014 19:01:35 +0000
Subject: [PATCH 0518/1077] MANTIS 0025026 change Mantis 0025026: Report
 process counters to logging service - extension in LoggingClass enum -
 modified LoggingCounterData class

git-svn-id: svn://localhost/SelfServiceCommon/trunk@642 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/idl/Massai/platform.idl | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index 361349b0..5ee6e7eb 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -191,8 +191,10 @@ module PlatformManagement
       LC_BIN2,
       LC_BIN3,
       LC_BIN4,
-      LC_CAPTURE,
-      LC_DISPENSER,
+      LC_CAPTURE_PRINTER,
+      LC_CAPTURE_READER,
+      LC_DISPENSER_PRINTER,
+	  LC_DISPENSER_READER,
       LC_PRINTER_FEEDER,
       LC_CONVEYOR_FEEDER,
       LC_DISK,
@@ -214,14 +216,15 @@ module PlatformManagement
     
     struct LoggingCounterData
     {
-      string realComponentName;
+	  string unitId;
       LoggingClass unitClass;
       long bin;                 //bin number
       long mediaStatusTypeValue;//int value of mediaStatusType
       long couponsproduced;
-      long numberOfDocs;
       long paperType;           //org.iata.cuss.Characteristics.MediaOutputPackage;
       string paperClass;        //com.materna.massai.DeviceComponents.characteristics.types
+      long mediaCount;
+      string type;              // type = "gpp" or type ="atb"; needed for evaluation of mediaStatusTypeValue
     };
     
     interface PlatformIF
@@ -230,7 +233,8 @@ module PlatformManagement
     typedef sequence<LoggingComponentData> LoggingComponentDataList;   
         void initLogging(in string realComponentName, in LoggingComponentDataList init);
         void logCounter(in LoggingCounterData lcd);
-        void logCounterFailure(in string realComponentName, in LoggingClass unitClass, in long numberOfFailed);
+        void logCounterFailure(in string unitId, in LoggingClass unitClass, in long numberOfFailed);
+        void updateMediaCount(in string compRef, in long mediaCount);
         
         /** Checks the status of the token passed by the application.
          *  @param token - The application token
-- 
2.41.0.windows.1

